utilities/tools/module-indexes/module-index.tests.ps1

```diff
@@ -0,0 +1,747 @@
+[CmdletBinding()]
+param (
+    [Parameter(Mandatory = $true, Position = 0, HelpMessage = 'Enter the full path to the CSV file.')]
+    [ValidateNotNullOrEmpty()]
+    [Alias('Path')]
+    [string]$CsvFilePath
+)
+
+BeforeAll {
+    $csvContent = Import-Csv -Path $CsvFilePath
+    $rawFile = Get-Content -Path $CsvFilePath
+    $csvHeaders = $csvContent[0].PSObject.Properties.Name
+
+    $issues = & gh issue list --limit 99999 --state all --json title,body | ConvertFrom-Json
```

Comment:
It might be a good call to switch this implmentation from `gh issue list` to `gh api` as it allows you to add the [--paginate](https://cli.github.com/manual/gh_api#:~:text=In%20%2D%2Dpaginate%20mode,outer%20JSON%20array.) parameter this will enable us to go beyond 1000 results. 

You can see an example of that [here](https://github.com/Azure/bicep-registry-modules/blob/1ae65ad3099a106bd6e9758f320b5ed3a26b4249/avm/utilities/tools/Invoke-WorkflowsForBranch.ps1#L161-L193) (just ignore the PAT if).

For small use cases the classic GH API commands lik `gh issue list` seem good enough. But for everything where we might end up with a large number of results, the `gh api` seems to be the better alternative.

The `$queryUrl` in your case should be `/issues`.

Thoughts?

Reply:
I'll look into this.

Reply:
Changed to gh api as suggested. Please take a look and resolve when ready. Thanks!

Reply:
Almost 😄 

You switched it, though I'm not sure what to think about the `{owner}`  & `{repo}`  strings in this url `"repos/{owner}/{repo}/issues?state=all&per_page=100"` - seem like placeholders. 
The bigger concern is however that you unfortunately added the option for pagination without actually implementing it. All that his 'should' return now, are 100 results. In the example I linked you can see that I actually loop through the pages in the loop and collect the results.
Did you test the cmdlet and it actually returns all results even without active pagination?